datasource/packer-image-iteration/data.go

```diff
@@ -0,0 +1,164 @@
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type DatasourceOutput,Config,ParBuild,ParImage
+package packer_image_iteration
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/hashicorp/hcl/v2/hcldec"
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-packer-service/preview/2021-04-30/models"
+	"github.com/hashicorp/packer-plugin-sdk/common"
+	"github.com/hashicorp/packer-plugin-sdk/hcl2helper"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+	"github.com/hashicorp/packer-plugin-sdk/template/config"
+	packerregistry "github.com/hashicorp/packer/internal/packer_registry"
+)
+
+type Datasource struct {
+	config Config
+}
+
+type Config struct {
+	common.PackerConfig `mapstructure:",squash"`
+	// The name of the bucket your image is in.
+	Bucket string `mapstructure:"bucket_name"`
+	// The name of the channel to use when retrieving your image
+	Channel string `mapstructure:"channel"`
+	// TODO: Version          string `mapstructure:"version"`
+	// TODO: Label          string `mapstructure:"label"`
+}
+
+func (d *Datasource) ConfigSpec() hcldec.ObjectSpec {
+	return d.config.FlatMapstructure().HCL2Spec()
+}
+
+func (d *Datasource) Configure(raws ...interface{}) error {
+	err := config.Decode(&d.config, nil, raws...)
+	if err != nil {
+		return err
+	}
+
+	var errs *packersdk.MultiError
+
+	if d.config.Bucket == "" {
+		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("The `bucket_name` must be specified"))
+	}
+	if d.config.Channel == "" {
+		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("`channel` is currently a required field."))
+	}
+
+	if errs != nil && len(errs.Errors) > 0 {
+		return errs
+	}
+	return nil
+}
+
+// Copy of []*models.HashicorpCloudPackerBuild. Need to copy so we can generate
+// the HCL spec.
+type ParBuild struct {
```

Comment:
I wonder if type aliasing would work here 🤔 : 

```suggestion
// Alias of models.HashicorpCloudPackerBuild. So we can generate the HCL spec.
type ParBuild = models.HashicorpCloudPackerBuild
```

Reply:
(to not have to update the field of this struct)